### PR TITLE
Feature/update docker build

### PR DESCRIPTION
**Pull request by aseefahmed**
## Description
Currently the docker build for our circulation manager backend needs some updates. Right now it builds pulling in the code from git within the docker build, it should instead be using the locally available files.

## Motivation and Context
- To avoid cloding duplicate codebase in every jobs

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docker/Dockerfile.exec [hunk 1]
```diff
@@ -24,14 +24,13 @@ ENV SIMPLIFIED_SCRIPT_NAME ""
 
 # Copy over all Palace build files for this image
 COPY . /ls_build
-
-RUN /bin/bash -c "/ls_build/simplified_app.sh ${repo} ${version} \
-      && /ls_build/logrotate.sh \
+RUN /bin/bash -c "/ls_build/docker/simplified_app.sh ${repo} ${version} \
```
Comment (jonathangreen) [inline docker/Dockerfile.exec:27]: Do `${repo}` and `${version}` do anything here now?

File: .github/workflows/test-build.yml [hunk 2]
```diff
@@ -259,7 +256,7 @@ jobs:
       - name: Push circ-webapp image
         uses: docker/build-push-action@v2
         with:
-          context: ./docker
+          context: .
```
Comment (jonathangreen) [inline .github/workflows/test-build.yml:259]: Do the build args below still have to be passed into this action?

File: docker/Dockerfile.exec [hunk 3]
```diff
@@ -24,14 +24,13 @@ ENV SIMPLIFIED_SCRIPT_NAME ""
...
+      && /ls_build/docker/logrotate.sh \
       && rm -rf /ls_build && /bd_build/cleanup.sh"
 
 VOLUME /var/log
 WORKDIR /home/simplified/circulation/bin
-
+RUN echo ${SIMPLIFIED_DB_TASK}
```
Comment (jonathangreen) [inline docker/Dockerfile.exec:33]: Is this line needed?

File: docker/Dockerfile.webapp [hunk 4]
```diff
@@ -10,10 +10,10 @@ ENV LANG=C.UTF-8
 # Copy over all Palace build files for this image
 COPY . /ls_build
```
Comment (jonathangreen) [inline docker/Dockerfile.webapp:11]: Instead of adding docker to the path everywhere, we could change this `COPY` command to be:
`COPY ./docker /lsbuild`

File: docker/simplified_app.sh [hunk 5]
```diff
@@ -40,10 +41,9 @@ export LANG=C.UTF-8
 useradd -ms /bin/bash -U simplified
 
 # Get the proper version of the codebase.
-mkdir /var/www && cd /var/www
-git clone https://github.com/${repo}.git circulation
-chown simplified:simplified circulation
-cd circulation
+mkdir -p /var/www/circulation && rsync -rtv /ls_build/ /var/www/circulation
```
Comment (jonathangreen) [inline docker/simplified_app.sh:44]: instead of doing an rsync here, couldn't we copy the files directly into this path: 
COPY . /var/www/circulation

**Review by jonathangreen — CHANGES_REQUESTED** [review]

**Comment by jonathangreen** [comment]
One more comment here, can you update the README file, since we no longer need the repo and version arguments.

**Comment by aseefahmed** [comment]
> 

I will do once tested.

File: docker/Dockerfile.webapp (hunk 4 above)
Comment (aseefahmed) [reply docker/Dockerfile.webapp:11]: i tested. it will not work. it should copy all the files including './docker'.

File: .github/workflows/test-build.yml [hunk 6]
```diff
@@ -74,6 +74,8 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+        with:
+          clean: false
```
Comment (jonathangreen) [inline .github/workflows/test-build.yml:78]: Why is this needed?

File: docker/Dockerfile.webapp (hunk 4 above)
Comment (jonathangreen) [reply docker/Dockerfile.webapp:8]: What happens in this case? Why won't it work?

**Review by jonathangreen — CHANGES_REQUESTED** [review]
A couple more comments here. I did test locally, and the image is working for me. Should be good to merge after these changes.

File: .github/workflows/test-build.yml [hunk 7]
```diff
@@ -130,9 +129,6 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - name: Set up Docker Buildx
```
Comment (jonathangreen) [inline .github/workflows/test-build.yml:133]: Why remove this action?

**Review by jonathangreen — CHANGES_REQUESTED** [review]

File: docker/Dockerfile.webapp (hunk 4 above)
Comment (aseefahmed) [reply docker/Dockerfile.webapp:8]: the whole project needs to be copied into /lsbuild, not just the docker directory

File: .github/workflows/test-build.yml (hunk 6 above)
Comment (aseefahmed) [reply .github/workflows/test-build.yml:78]: actually, its not needed, i will remove those lines.

File: docker/Dockerfile.webapp (hunk 4 above)
Comment (jonathangreen) [reply docker/Dockerfile.webapp:8]: I don't understand why this is the case. We did not copy the whole project into lsbuild before, why is it needed now?

Comment (aseefahmed) [reply docker/Dockerfile.webapp:8]: ok. i will do double check if i miss anything

File: .github/workflows/test-build.yml (hunk 7 above)
Comment (aseefahmed) [reply .github/workflows/test-build.yml:133]: you right, it shouldn't be removed. i'll undo.

**Comment by jonathangreen** [comment]
@aseefahmed I pushed a commit here to not copy everything into the `ls_build` folder. If you can fix those other issues, then I will test again. I think this is close to ready to merge.

File: docker/Dockerfile.webapp (hunk 4 above)
Comment (aseefahmed) [reply docker/Dockerfile.webapp]: seems working now.

**Review by jonathangreen — APPROVED** [review]
Tested this locally and it looks good
